### DISCUSSION TITLE
Deploying docker image with Nginx

### DISCUSSION THREAD

**Discussion by kleysonr** [tiangolo/uvicorn-gunicorn-docker#229] [Questions] [answered]
The Gunicorn documentation says to have an Nginx in front of the Gunicorn.

How to change the uvicorn-gunicorn-docker to add also the Nginx to handle the requests ?

**Comment by mawoka-myblock** [comment]
Never expose any container directly to the internet. You should use a Reverse Proxy. That's also what they mean in the docs,

**EDIT**:
I meant that you shouldn't expose any containers which aren't made for being exposed.

**Comment by NargiT** [comment]
if you have docker you can use another container to act as a proxy.
in kubernetes with an ingress, in docker compose with another another docker (apache or nginx) will do this job if you want to expose it outside. 

But you should not stack layer inside a docker container since it's suppose to be elastic. with an nginx inside you are no more elastic.

**Comment by tiangolo** [comment, marked as answer]
Now that Uvicorn supports managing workers with `--workers`, including restarting dead ones, there's no need for Gunicorn. That also means that it's much simpler to build a Docker image from scratch now, I updated the docs to explain it.

Because of that, I deprecated this Docker image: https://github.com/tiangolo/uvicorn-gunicorn-docker#-warning-you-probably-dont-need-this-docker-image

You can probably read more about all that process to have a termination proxy in the FastAPI docs for Deployments with Docker: https://fastapi.tiangolo.com/deployment/docker/
